### PR TITLE
feat: pr comment generator for covector preview

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation

To finish off the works of #179 and #187. 

## Approach
- When configuring the preview workflow, user will now be required to pass in a token:
  ```yaml
  - name: covector preview
    uses: jbolda/covector/packages/action@main
    id: covector
    with:
      command: preview
      token: ${{ secrets.MY_TOKEN }}
  ```
- Comment contents:
  - no preview packages = `Covector did not publish any preview packages`
  - yes preview packages = `The following preview packages have been published by Covector: ...` (see real example below)
    - We should be able to get our own bot to create the comment (instead of `github-action` bot) if we pass in the bot's token in the workflow.
- The covector action will rewrite its previous comment in a given PR as long as it detects that there's only one covector-generated comment. If there is none or more than one, it will create new comments instead of trying to figure out which comment to edit.
- Added change file for `action` patch

## TODOs
- [x] Merge #187 and #193 first
- [x] ~Typescript could use some work...~ L8R
- [x] ~It's already been mentioned in #187 but the boolean that determines if a package has been [published or not](https://github.com/jbolda/covector/blob/main/packages/action/src/index.ts#L80-L87) is never toggled (I think it's just one of those things that's been overlooked). Covector will rely on this boolean for generating its PR comments.~ Separate PR
- [x] ~See example of comment template down below. We could fancy it up with a table if you'd like. I like having it as one string so it's easier to copy and paste to the terminal.~ Separate PR

### CODE REVIEW

**Comment by minkimcello** [comment]
aldkfjawoeifja1234adfjbbalkjefnb

awepfijopaoje2345afiopjefaf

beeopbobop

**Comment by minkimcello** [comment]
apweojfalk9999aowiejfoiaj

anlknaewfa8888awoiefjoiaj

aewpofjapeojf7777aweoifjaw

**Comment by jbolda** [comment]
> Typescript could use some work...

I know that feeling 😄 

> It's already been mentioned in #187 but the boolean that determines if a package has been published or not is never toggled (I think it's just one of those things that's been overlooked). Covector will rely on this boolean for generating its PR comments.

Oh I like that thought. Perhaps we should address this issue in this PR? We will likely want to add some tests around it to confirm that we are actually returning properly. Particularly in case something inside covector changes, we don't accidentally break comments. We could probably mock out the comment generation much like I mocked out the Github Release to confirm that our inputs are indeed correct.

> See example of comment template down below. We could fancy it up with a table if you'd like. I like having it as one string so it's easier to copy and paste to the terminal.

I think I agree with all of this. Perhaps we decide to adjust in the future, but I think this is a great as is.

**Comment by jbolda** [comment]
One question that I thought of this morning... Should we update / keep the first comment up to date with the released version? I think you haven't, and I also feel like the answer should be "no". However, I just wanted to explicitly rule it out if you agree.

**Comment by minkimcello** [comment]
> Should we update / keep the first comment up to date with the released version?

What do you mean?

**Comment by jbolda** [comment]
> > Should we update / keep the first comment up to date with the released version?
> 
> What do you mean?

The bot could create a message initially, and after each commit edit and update the first message with the newly published packages. As opposed to adding a new comment every time a new publish happens.

**Comment by minkimcello** [comment]
> > > Should we update / keep the first comment up to date with the released version?
> > 
> > 
> > What do you mean?
> 
> The bot could create a message initially, and after each commit edit and update the first message with the newly published packages. As opposed to adding a new comment every time a new publish happens.

Oh yeah it does that right now. I tag the covector-generated comment with a hidden `<!--covector-->` (or something like that). So each time it'll check to see if this comment exists, if so, edit that comment, if not, create a new comment.

**Comment by jbolda** [comment]
Oh cool perfect 👍

File: .changes/preview-comments.md
Comment (jbolda) [inline .changes/preview-comments.md:2]: We can call this a minor as well. I would qualify comments as a big enough feature.
```suggestion
"action": minor
```

**Review by jbolda — APPROVED** [review]
Cheers! Feel free to merge this after we get #187 merged in.

**Pull request merged**
